sdk/servicebus/Microsoft.Azure.ServiceBus/src/MessageReceivePump.cs

```diff
@@ -86,8 +86,9 @@ async Task MessagePumpTaskAsync()
                         }
                         catch (Exception exception)
                         {
-                            // Not reporting an ObjectDisposedException as we're stopping the pump
-                            if (!(exception is ObjectDisposedException && this.pumpCancellationToken.IsCancellationRequested))
+                            // Not reporting an ObjectDisposedException and OperationCanceledException as we're stopping the pump
+                            if (!(this.pumpCancellationToken.IsCancellationRequested && 
```

Comment:
nit:  It may be cleaner and more readable to move this to its own catch block with an exception filter, so that the expected and exceptional cases are segmented.

Reply:
Fixed